include/deal.II/distributed/tria_base.h

```diff
@@ -132,6 +132,26 @@ namespace parallel
      */
     types::subdomain_id locally_owned_subdomain () const;
 
+    /**
+     * Returns a set of MPI ranks of the processors that have at least one
+     * ghost cell adjacent to the cells of the local processor. In other
+     * words, this is the set of subdomain_id() for all ghost cells.
+     *
+     * @note: If @p i is contained in the list of processor @p j, then @p j
+     * will also be contained in the list of processor @p i.
+     **/
+    virtual const std::set<unsigned int> &ghost_owners () const;
+
+    /**
+     * Returns a set of MPI ranks of the processors that have at least one
+     * level ghost cell adjacent to our cells used in geometric multigrid. In
+     * other words, this is the set of level_subdomain_id() for all level
+     * ghost cells.
+     *
+     * @note: If @p i is contained in the list of processor @p j, then @p j
+     * will also be contained in the list of processor @p i.
+     **/
+    virtual const std::set<unsigned int> &level_ghost_owners () const;
```

Comment:
I admit that I implemented exactly what I currently needed. Do you want me to go with the by level implementation? Or both?